dm/ctl/master/query_status.go

```diff
@@ -60,5 +74,49 @@ func queryStatusFunc(cmd *cobra.Command, _ []string) {
 		return
 	}
 
-	common.PrettyPrintResponse(resp)
+	if resp.Result && taskName == "" && len(workers) == 0 {
+		result := wrapTaskResult(resp)
+		common.PrettyPrintInterface(result)
+	} else {
+		common.PrettyPrintResponse(resp)
+	}
+}
+
+// wrapTaskResult pick task info and generate tasks' status and relative workers
+func wrapTaskResult(resp *pb.QueryStatusListResponse) *taskResult {
+	taskStatusMap := make(map[string]string, len(resp.Workers))
+	taskCorrespondingWorkers := make(map[string][]string, len(resp.Workers))
+	for _, worker := range resp.Workers {
+		for _, subTask := range worker.SubTaskStatus {
+			subTaskName := subTask.Name
+			subTaskStage := subTask.Stage
+
+			taskCorrespondingWorkers[subTaskName] = append(taskCorrespondingWorkers[subTaskName], worker.Worker)
+			taskStage := taskStatusMap[subTaskName]
+			switch {
+			case taskStage == stageError:
+			case subTaskStage == pb.Stage_Paused && subTask.Result != nil && len(subTask.Result.Errors) > 0:
+				taskStatusMap[subTaskName] = stageError
+			case taskStage == pb.Stage_Paused.String():
```

Comment:
1. do nothing for this `case`? if so, can we remove it?

2. should we add some comments to explain these stages here?

Reply:
1. If this case is removed, taskStage might be set to "Running" although taskStage is "Paused".

2. I will add some comments.